### PR TITLE
Sparkline optimisations

### PR DESCRIPTION
**Pull request by jennygrahamjones**
## What's changed?

Instead of re-rendering the collection every time new page view data is received from Ophan, only the article graph itself is re-rendered. 

Additionally, the call to Ophan will now only be made if the user has toggled on the page view data feature. 

### General
- [ ] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [x] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [x] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

File: client-v2/src/redux/modules/pageViewData/selectors.ts [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
-  );
+): PageViewStory | undefined => {
+  const dataForFront: PageViewDataPerFront | undefined = selectPageViewData(
+    state
+  ).find(front => front.frontId === frontId);
+
+  const allArticlesOnFront =
+    dataForFront &&
+    dataForFront.collections.reduce(
+      (acc: PageViewArticlesOnFront, curr: PageViewDataPerCollection) => {
```
Comment (jonathonherbert) [inline client-v2/src/redux/modules/pageViewData/selectors.ts]: We'll need to memoize this to ensure we're not doing expensive graph work unnecessarily -- happy to pair!

File: client-v2/src/shared/components/article/ArticleGraph.tsx [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
+          />
+        </AreaChart>
+      </>
     );
   }
 }
 
-export default ArticleGraph;
+const mapStateToProps = (state: State, props: ArticleGraphProps) => ({
+  data: selectPageViewDataForArticleId(state, props.articleId, props.frontId)
```
Comment (jonathonherbert) [inline client-v2/src/shared/components/article/ArticleGraph.tsx]: Awesome 👌

File: client-v2/src/util/pollingConfig.ts [hunk 3]
```diff
@@ -24,6 +24,12 @@ export default (store: Store) =>
     }
     (store.dispatch as Dispatch)(fetchStaleOpenCollections(priority));
 
+    if (
```
Comment (jonathonherbert) [inline client-v2/src/util/pollingConfig.ts]: 👍

**Comment by ajwl** [comment]
@jonathonherbert thanks

**Pull request merged**